orangecontrib/text/corpus.py

```diff
@@ -73,12 +85,14 @@ def __init__(self, domain=None, X=None, Y=None, metas=None, W=None,
                 text mining. Infer them if None.
             ids (numpy.ndarray): Indices
         """
+        super().__init__()
         n_doc = _check_arrays(X, Y, metas)
 
-        self.X = X if X is not None else np.zeros((n_doc, 0))
-        self.Y = Y if Y is not None else np.zeros((n_doc, 0))
-        self.metas = metas if metas is not None else np.zeros((n_doc, 0))
-        self.W = W if W is not None else np.zeros((n_doc, 0))
+        with self.unlocked():
```

Comment:
I would use unlocked_reference here, because it is enough

Reply:
Done.